### PR TITLE
Javascript javascript1 week2/guzide

### PR DESCRIPTION
**Pull request by GuzideGuzelbey**
Tried to complete the optional task also, not sure if it is doing exactly what it should do :)

### CODE REVIEW

File: javascript/javascript1/week1/week1-guzide/trialHouseyPriceyWfunction.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
+const customerArray = [
+  (peterHouse = {
```
Comment (julczka) [inline javascript/javascript1/week1/week1-guzide/trialHouseyPriceyWfunction.js:2]: This causes an error in the console, please test your code before submitting. You can use https://stackblitz.com/ for quick tests. It allows you to spin up a development environment with different configurations - for now you can choose vanilla/static

File: javascript/javascript1/week2/week2-guzide/candy-helper/CandyHelper.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    return true;
+  } else {
+    console.log("Enough candy for you!");
+    return false;
+  }
+}
+
+addCandy("toffee", 40);
+addCandy("sweet", 61);
+addCandy("chewing-gum");
```
Comment (julczka) [inline javascript/javascript1/week2/week2-guzide/candy-helper/CandyHelper.js]: Here you are not passing a second arument to your `addCandy` function. That causes a bug in your code. Run it and try to log total price after the loop from line 32 is executed.

File: javascript/javascript1/week2/week2-guzide/event-application/EventApp.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  "Wednesday",
+  "Thursday",
+  "Friday",
+  "Saturday",
+];
+
+function getEventWeekday(daysFromNow) {
+  const todaysIndex = new Date().getDay();
+  const eventDaysIndex = (todaysIndex + daysFromNow) % 7;
+  return console.log(weekDaysArr[eventDaysIndex]);
```
Comment (julczka) [inline javascript/javascript1/week2/week2-guzide/event-application/EventApp.js]: Very nice use of modulus!
You do not need the keyword `return` before console.log. It would make more sense to return the actual weekday from this function, then the value can be stored in a varialbe and the fuction will be more reusable. Try to refactor the code that way.

File: javascript/javascript1/week2/week2-guzide/flight-booking/FligthBooking.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
+function getFullname(
+  firstname,
+  surname,
+  useFormalName = false,
+  gender = "female"
+) {
+  firstname = firstname || "";
```
Comment (julczka) [inline javascript/javascript1/week2/week2-guzide/flight-booking/FligthBooking.js]: I think you do not need these two lines, if the intention was to test if it is an empty string then line 10 does that.

File: javascript/javascript1/week2/week2-guzide/flight-booking/FligthBooking.js [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+  surname = surname || "";
+
+  if (!firstname && !surname) {
```
Comment (julczka) [inline javascript/javascript1/week2/week2-guzide/flight-booking/FligthBooking.js]: Should that be `&&` ? We know the funcion cannot wokr if we do not have both first name and surname, but can it work correctly if we have only one of those?

File: javascript/javascript1/week2/week2-guzide/flight-booking/FligthBooking.js [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+    if (gender === "female") {
+      nameTitle = "Lady ";
+    } else if (gender === "male") {
+      nameTitle = "Lord ";
+    } else {
+      nameTitle = "";
+    }
+  }
+
+  return console.log(nameTitle + firstname + " " + surname);
```
Comment (julczka) [inline javascript/javascript1/week2/week2-guzide/flight-booking/FligthBooking.js]:
```suggestion
const fullName = nameTitle + firstname + " " + surname
  return fullName;
```

Here also you can return the compujted value from the funcion and then in the script you can write `console.log(getFullname("Benjamin", "Hughes", true, ""))` to see the value that funcion returns

File: javascript/javascript1/week2/week2-guzide/student-manager/StudentManager.js [hunk 7]
```diff
@@ -0,0 +1,37 @@
+const class07Students = [];
+
+function getNumberOfStudents() {
+  return class07Students.length;
+}
+
+function addStudentToClass(studentName) {
+  if (studentName == "") {
+    console.log("You can't leave this area empty");
```
Comment (julczka) [inline javascript/javascript1/week2/week2-guzide/student-manager/StudentManager.js]: It is good to write explicit and clear error messages - I think something like "Student name cannot be empty" would work better. It is bit hard to say what "this area" refers to specifically.

File: javascript/javascript1/week2/week2-guzide/student-manager/StudentManager.js [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+    return;
+  }
+
+  if (studentName === "Queen Mary") {
```
Comment (julczka) [inline javascript/javascript1/week2/week2-guzide/student-manager/StudentManager.js]: I think the specification for this just mentioned "Quenn" as a special student name :)

File: javascript/javascript1/week2/week2-guzide/student-manager/StudentManager.js [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+  if (getNumberOfStudents() >= 6) {
+    console.log("Cannot add more students to class 07");
+    return;
+  }
+  class07Students.push(studentName);
+  console.log(class07Students);
+}
+
+addStudentToClass("");
+addStudentToClass("Güzide");
```
Comment (julczka) [inline javascript/javascript1/week2/week2-guzide/student-manager/StudentManager.js:32]: You have a bug in your code 🐛 
You need to move one if statement to make it work properly. Try adding the same name twice to the students array.

File: javascript/javascript1/week2/week2-guzide/weather-wear/WeatherWear.js [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+      console.log("Wear 2 layers; outer layer can be a coat");
+      break;
+    case temperature >= 10 && temperature < 20:
+      console.log("Wear 2 layers; outer layer can be a jacket");
+      break;
+    case temperature >= 20 && temperature < 30:
+      console.log("Wear 1 thin layer");
+      break;
+    default:
+      console.log("You don't have to wear anything!");
```
Comment (julczka) [inline javascript/javascript1/week2/week2-guzide/weather-wear/WeatherWear.js:16]: That is a brave choice :)

File: javascript/javascript1/week2/week2-guzide/weather-wear/WeatherWear.js [hunk 11]
```diff
@@ -0,0 +1,22 @@
+function chooseClothes(temperature) {
+  switch (true) {
```
Comment (julczka) [inline javascript/javascript1/week2/week2-guzide/weather-wear/WeatherWear.js:2]: I think there is a mistake in this line :)
try to test the code for each case, this fuction does not work as intended even witn the values  you test for.

**Review by julczka — CHANGES_REQUESTED** [review]
Hey, there are few bugs here and there. Try to test your code thoroughly before submitting - if there is a requirement written in the homework, for example that you cannot add duplicated students to the class you should try to do that to see if your code works correctly.

File: javascript/javascript1/week2/week2-guzide/candy-helper/CandyHelper.js (hunk 2 above)
Comment (GuzideGuzelbey) [reply javascript/javascript1/week2/week2-guzide/candy-helper/CandyHelper.js]: Thank you Julia. The issue is resolved.

File: javascript/javascript1/week2/week2-guzide/event-application/EventApp.js (hunk 3 above)
Comment (GuzideGuzelbey) [reply javascript/javascript1/week2/week2-guzide/event-application/EventApp.js]: The function is refactored.

File: javascript/javascript1/week2/week2-guzide/flight-booking/FligthBooking.js (hunk 4 above)
Comment (GuzideGuzelbey) [reply javascript/javascript1/week2/week2-guzide/flight-booking/FligthBooking.js]: Yeaps you are right. Extra lines for checking if the input is an empty string is removed.

File: javascript/javascript1/week2/week2-guzide/flight-booking/FligthBooking.js (hunk 6 above)
Comment (GuzideGuzelbey) [reply javascript/javascript1/week2/week2-guzide/flight-booking/FligthBooking.js]: It helped me a lot to solve the previous 2 issues as well. Thank you for the feedback.

File: javascript/javascript1/week2/week2-guzide/flight-booking/FligthBooking.js (hunk 5 above)
Comment (GuzideGuzelbey) [reply javascript/javascript1/week2/week2-guzide/flight-booking/FligthBooking.js]: Yes definitely, by using `||` problem is resolved.

File: javascript/javascript1/week2/week2-guzide/student-manager/StudentManager.js (hunk 7 above)
Comment (GuzideGuzelbey) [reply javascript/javascript1/week2/week2-guzide/student-manager/StudentManager.js]: Yes my bad, thank you.

File: javascript/javascript1/week2/week2-guzide/student-manager/StudentManager.js (hunk 8 above)
Comment (GuzideGuzelbey) [reply javascript/javascript1/week2/week2-guzide/student-manager/StudentManager.js]: `"Queen Mary"`is changed with `"Queen"`.

File: javascript/javascript1/week2/week2-guzide/student-manager/StudentManager.js (hunk 9 above)
Comment (GuzideGuzelbey) [reply javascript/javascript1/week2/week2-guzide/student-manager/StudentManager.js:32]: Issue is resolved. Thank you.

File: javascript/javascript1/week2/week2-guzide/weather-wear/WeatherWear.js (hunk 10 above)
Comment (GuzideGuzelbey) [reply javascript/javascript1/week2/week2-guzide/weather-wear/WeatherWear.js:16]: :D

File: javascript/javascript1/week2/week2-guzide/weather-wear/WeatherWear.js (hunk 11 above)
Comment (GuzideGuzelbey) [reply javascript/javascript1/week2/week2-guzide/weather-wear/WeatherWear.js:2]: I double checked each condition (5 conditions) and couldn't find any problem.
for these inputs: `chooseClothes(5);
chooseClothes(40);
chooseClothes(15);
chooseClothes(28);
chooseClothes(-40);` I get those ouptputs: `Wear 2 layers; outer layer can be a coat
You don't have to wear anything!
Wear 2 layers; outer layer can be a jacket
Wear 1 thin layer
Wear at least 3 layers and use at least 1 termal layer`

**Comment by GuzideGuzelbey** [comment]
First of all, I wanna apologise for the late response (I had PD3 exam in that period, could only focus on homeworks each week then jump to the next task). 
Thank you for your constructive feedbacks :)
